portal/static/js/profile.js

```diff
@@ -433,6 +437,15 @@
             isAdmin: function() {
                 return this.currentUserRoles.indexOf("admin") !== -1;
             },
+            isPatient: function() {
+                if (this.mode === "createPatientAccount" || $("#profileMainContent").hasClass("patient-view")) {
+                    return true;
+                } 
+                if (this.userRoles.length === 0) { //this is a blocking call if not cached, so avoid it if possible
```

Comment:
Shouldn't the role data be cached in localStorage by a service worker at this point?

Reply:
@ivan-c ivan, you are right, it will be if the call to get role data is made, but it is not always made (e.g. as in account creation page)

Reply:
Ah, good to know.